src/main/java/fi/metatavu/metamind/bot/functions/KuntaApiGetEventsMetaBotFunction.java

```diff
@@ -0,0 +1,184 @@
+      story.getVariablesState().set("eventDescription", events.get(rand).getDescription());
+      story.getVariablesState().set("eventLocation", events.get(rand).getPlace());
+      story.getVariablesState().set("eventAddress", events.get(rand).getAddress());
+      story.getVariablesState().set("eventZip", events.get(rand).getZip());
+      story.getVariablesState().set("eventCity", events.get(rand).getCity());
+    } catch (Exception e) {
+      logger.error("Error while setting variables at KUNTA_API_GET_EVENTS", e);
+    }
+  }
+
+  /**
+   * This method returns a HttpURLConnection based on organizationId, route and
+   * routeId.
+   * 
+   * @param organizationId Organization id of kunta-api oranization.
+   * @param route String The route where to get data.
+   * @param routeId (Id or blank String) Route id to get specific data by id.
+   * @return A HttpURLConnection.
+   */
+  private HttpURLConnection createHttpConnection(String authorizationToken, String organizationId, String route,
+      String routeId) {
+    URL url = null;
+
+    try {
+      url = new URL(String.format("http://api.kunta-api.fi/v1/organizations/%s/%s/%s", organizationId, route, routeId));
+    } catch (NullPointerException | MalformedURLException e) {
+      logger.error("Failed to create URL at KUNTA_API_GET_EVENTS");
+    }
+
+    HttpURLConnection connection = null;
+
+    try {
+      if (url != null) {
+        connection = (HttpURLConnection) url.openConnection();
+      }
+    } catch (NullPointerException | IOException e) {
+      logger.error("Failed to create connection at KUNTA_API_GET_EVENTS", e);
+    }
+
+    String redirect = connection.getHeaderField("Location");
+
+    if (redirect != null) {
+      try {
+        connection = (HttpURLConnection) new URL(redirect).openConnection();
+      } catch (NullPointerException | IOException e) {
+        logger.error("Failed to set redirect URL at KUNTA_API_GET_EVENTS", e);
+      }
+    }
+
+    try {
+      connection.setRequestMethod("GET");
+    } catch (NullPointerException | ProtocolException e) {
+      logger.error("Failed to set request method GET at KUNTA_API_GET_EVENTS", e);
+    }
+
+    connection.setRequestProperty("Content-Type", "application/json");
+    connection.setRequestProperty("Authorization", authorizationToken);
+
+    return connection;
+  }
+
+  /**
+   * This method returns a StringBuilder content from given connection.
+   * 
+   * @param connection HttpURLConnection.
+   * @return A StringBuilder content based on connection.
+   */
+  private StringBuilder getContent(HttpURLConnection connection) {
+    BufferedReader input = null;
+
+    try {
+      input = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+    } catch (NullPointerException | IOException e) {
+      logger.error("Failed to create BufferReader at KUNTA_API_GET_EVENTS", e);
+    }
+
+    String inputLine;
+    StringBuilder content = new StringBuilder();
+
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "input" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)